### PR TITLE
Refactor: wrap rocksdb transaction operations in closures

### PR DESCRIPTION
**Pull request by imtsuki**
## Description

Wraps most of the redundant rocksdb transaction operations in two new helper functions, `with_txn` and `transactional_update_or_else`.

This is the first step of extracting rocksdb logic and adding a db-agnostic layer for client-storage.

### CODE REVIEW

File: dragonfly-client-storage/src/metadata.rs [hunk 1]
```diff
@@ -46,12 +46,18 @@ const DEFAULT_BLOCK_SIZE: usize = 64 * 1024;
...
-// TASK_CF_NAME is the column family name of task.
+/// The column family name of [Task].
 const TASK_CF_NAME: &str = "task";
 
-// PIECE_CF_NAME is the column family name of piece.
+/// The column family name of [Piece].
 const PIECE_CF_NAME: &str = "piece";
 
+/// Trait that marks a type can be stored in rocksdb, which has a cf name.
+trait RocksdbStorable: Default + Serialize + for<'de> Deserialize<'de> {
```
Comment (gaius-qi) [inline dragonfly-client-storage/src/metadata.rs:56]: I think it should be named similar to `ColumnFamilyDescriptor`.

File: dragonfly-client-storage/src/metadata.rs [hunk 2]
```diff
@@ -46,12 +46,18 @@ const DEFAULT_BLOCK_SIZE: usize = 64 * 1024;
 // DEFAULT_CACHE_SIZE is the default cache size for rocksdb.
 const DEFAULT_CACHE_SIZE: usize = 16 * 1024 * 1024;
 
-// TASK_CF_NAME is the column family name of task.
+/// The column family name of [Task].
```
Comment (gaius-qi) [inline dragonfly-client-storage/src/metadata.rs:49]: The style of code comments should be consistent, with variable or method names positioned at the beginning.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Comment by gaius-qi** [comment]
Ignore the CI results.

File: dragonfly-client-storage/src/metadata.rs (hunk 2 above)
Comment (imtsuki) [reply dragonfly-client-storage/src/metadata.rs:49]: I think Rust's docstring convention does not require this, this looks more like Go's comment style; but I agree that this change should be further discussed and probably be addressed in a separate PR which will change the code comment style for the whole project to keep consistency.

File: dragonfly-client-storage/src/metadata.rs [hunk 3]
```diff
@@ -46,12 +46,18 @@ const DEFAULT_BLOCK_SIZE: usize = 64 * 1024;
...
-// TASK_CF_NAME is the column family name of task.
+/// TASK_CF_NAME is the column family name of [Task].
 const TASK_CF_NAME: &str = "task";
 
-// PIECE_CF_NAME is the column family name of piece.
+/// PIECE_CF_NAME is the column family name of [Piece].
 const PIECE_CF_NAME: &str = "piece";
 
+/// ColumnFamilyDescriptor marks a type can be stored in rocksdb, which has a cf name.
+trait ColumnFamilyDescriptor: Default + Serialize + for<'de> Deserialize<'de> {
```
Comment (xujihui1985) [inline dragonfly-client-storage/src/metadata.rs:56]: can use DeserializeOwned if doesn't borrow any data from Deserializer. actually DeserializeOwned:for<'de> Deserialize<'de>

Comment (imtsuki) [reply dragonfly-client-storage/src/metadata.rs]: done

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**